tests/data/fields/text_field_test.py

```diff
@@ -88,19 +88,19 @@ def test_index_converts_field_correctly(self):
         t_index = vocab.add_token_to_namespace("t", namespace='characters')
         c_index = vocab.add_token_to_namespace("c", namespace='characters')
 
-        field = TextField(["A", "sentence"], [token_indexers["single id"](token_namespace="words")])
+        field = TextField(["A", "sentence"], [token_indexers["single id"](namespace="words")])
         field.index(vocab)
         # pylint: disable=protected-access
         assert field._indexed_tokens == [[capital_a_index, sentence_index]]
```

Comment:
I'm wondering if directly linking those two is desirable, and I'm thinking it's not (I thought you also had some concerns about this when we first talked about it).  The vocab namespace is something that you shouldn't have to think about at all except for rare edge cases, but you have to if these are linked.  If we instead just have arbitrary keys, it's really easy to make sure they match up in the configuration file:

```json
{
  "token_indexers": {
    "words": "single_token",
    "characters": "token_characters",
  },
  ...
  "token embedder": {
    "words": "single_token_embedding",
    "characters": {
      "type": "cnn"
    }
  }
}
```

(some details missing, but I think you get the idea)

If you don't use a dictionary to do this, the keys that a token embedder should look for look like magic.